shared/js/background/chrome-events.es6.js

```diff
@@ -38,6 +45,111 @@ chrome.runtime.onInstalled.addListener(function (details) {
     }
 })
 
+/**
+ * ONBOARDING
+ * Logic to allow the SERP to display onboarding UI
+ */
+
+/**
+  * !! This function _can_ be sync (`cb` is called in the same tick) if the settings are alrady loaded
+  */
+function getOnboardingParams (cb) {
+    function onReady () {
+        cb(null, {
+            showWelcomeBanner: settings.getSetting('showWelcomeBanner'),
+            showCounterMessaging: settings.getSetting('showCounterMessaging')
+        })
+    }
+
+    if (settings.isReady()) {
+        onReady()
+    } else {
+        settings.ready().then(onReady)
+    }
+}
+
+let onBeforeNavigateTimeStamp = null
+chrome.webNavigation.onBeforeNavigate.addListener(details => {
+    onBeforeNavigateTimeStamp = details.timeStamp
+})
+
+chrome.webNavigation.onCommitted.addListener(details => {
+    if (details.url.includes('duckduckgo.com/?')) {
```

Comment:
Can you make this a [url filter](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/events/UrlFilter) instead?

The performance improvement might not be significant but it should reduce risk also.

Reply:
Thanks, I didn't know about `URLFilter`. This should now be fixed 👍